### PR TITLE
[shellcheck SC2086] Double quote to prevent globbing and word splitting 

### PR DESCRIPTION
**Pull request by linleybao**
shellcheck SC2086 fix
https://github.com/koalaman/shellcheck/wiki/SC2086
shellcheck SC2004 fix
https://github.com/koalaman/shellcheck/wiki/SC2004

### CODE REVIEW

File: rsync_tmbackup.sh [hunk 1]
```diff
@@ -32,230 +32,232 @@ trap 'fn_terminate_script' SIGINT
...
+            # Because under MacOS X Tiger 'date -j' doesn't work, we do this:
+            perl -e 'use Time::Local; print timelocal('${ss}','${mi}','${hh}','${dd}','${mm}','${yy}'),"\n";' ;;
+        darwin*) date -j -f "%Y-%m-%d-%H%M%S" "$1" "+%s" ;;
+        FreeBSD*) date -j -f "%Y-%m-%d-%H%M%S" "$1" "+%s" ;;
+    esac
 }
 
 fn_find_backups() {
-	fn_run_cmd "find "$DEST_FOLDER/" -maxdepth 1 -type d -name \"????-??-??-??????\" -prune | sort -r"
+    fn_run_cmd "find \"$DEST_FOLDER/\" -maxdepth 1 -type d -name \"????-??-??-??????\" -prune | sort -r"
```
Comment (SimonHeimberg) [inline rsync_tmbackup.sh]: put `$DEST_FOLDER` in single quotes instead `of \"`

File: rsync_tmbackup.sh [hunk 2]
```diff
@@ -32,230 +32,232 @@ trap 'fn_terminate_script' SIGINT
...
-	if [ -z "$(fn_find_backup_marker "$(dirname -- "$1")")" ]; then
-		fn_log_error "$1 is not on a backup destination - aborting."
-		exit 1
-	fi
-
-	fn_log_info "Expiring $1"
-	fn_rm_dir "$1"
+    # Double-check that we're on a backup destination to be completely
+    # sure we're deleting the right folder
+    if [[ -z "$(fn_find_backup_marker "$(dirname -- "$1")")" ]]; then
```
Comment (SimonHeimberg) [inline rsync_tmbackup.sh]: change unrelated to this commit (tab to spaces)

any why this change "[[" instead of "["? [Shellcheck.net](shellcheck.net) does not give a hint to do this.

File: rsync_tmbackup.sh [hunk 3]
```diff
@@ -32,230 +32,232 @@ trap 'fn_terminate_script' SIGINT
...
+        cygwin*) date -d "${1:0:10} ${1:11:2}:${1:13:2}:${1:15:2}" +%s ;;
+        netbsd*) date -d "${1:0:10} ${1:11:2}:${1:13:2}:${1:15:2}" +%s ;;
+        darwin8*) yy=`expr ${1:0:4}`
+            mm=`expr ${1:5:2} - 1`
+            dd=`expr ${1:8:2}`
+            hh=`expr ${1:11:2}`
+            mi=`expr ${1:13:2}`
+            ss=`expr ${1:15:2}`
+            # Because under MacOS X Tiger 'date -j' doesn't work, we do this:
+            perl -e 'use Time::Local; print timelocal('${ss}','${mi}','${hh}','${dd}','${mm}','${yy}'),"\n";' ;;
```
Comment (SimonHeimberg) [inline rsync_tmbackup.sh]: change unrelated to this commit (tab to spaces)

File: rsync_tmbackup.sh [hunk 4]
```diff
@@ -360,28 +362,28 @@ DEST_FOLDER="${DEST_FOLDER%/}"
...
 
-if [ -n "$SSH_DEST_FOLDER" ]; then
-	DEST_FOLDER="$SSH_DEST_FOLDER"
+if [[ -n "$SSH_DEST_FOLDER" ]]; then
+    DEST_FOLDER="$SSH_DEST_FOLDER"
 fi
 
-if [ -n "$SSH_SRC_FOLDER" ]; then
-	SRC_FOLDER="$SSH_SRC_FOLDER"
+if [[ -n "$SSH_SRC_FOLDER" ]]; then
```
Comment (SimonHeimberg) [inline rsync_tmbackup.sh]: change unrelated to this commit (tab to spaces)

File: rsync_tmbackup.sh [hunk 5]
```diff
@@ -438,185 +440,185 @@ MYPID="$$"
 # Create log folder if it doesn't exist
 # -----------------------------------------------------------------------------
 
-if [ ! -d "$LOG_DIR" ]; then
-	fn_log_info "Creating log folder in '$LOG_DIR'..."
-	mkdir -- "$LOG_DIR"
+if [[ ! -d "$LOG_DIR" ]]; then
```
Comment (SimonHeimberg) [inline rsync_tmbackup.sh]: change unrelated to this commit (tab to spaces)

File: rsync_tmbackup.sh [hunk 6]
```diff
@@ -32,230 +32,230 @@ trap 'fn_terminate_script' SIGINT
...
-					# No: Keep it.
-					# this is now the last kept backup
-					last_kept_timestamp=$backup_timestamp
-					# and go to the next backup
-					break
-				fi
-			fi
-		done
-	done
+    local current_timestamp=${EPOCH}
```
Comment (SimonHeimberg) [inline rsync_tmbackup.sh]: change unrelated to this commit (tab to spaces)

File: rsync_tmbackup.sh [hunk 7]
```diff
@@ -32,230 +32,230 @@ trap 'fn_terminate_script' SIGINT
...
-		fi
-		SSH_SRC_FOLDER_PREFIX="${SSH_USER}@${SSH_HOST}:"
-	fi
+    # To keep compatibility with bash version < 3, we use grep
+    if echo "$DEST_FOLDER"|grep -Eq '^[A-Za-z0-9\._%\+\-]+@[A-Za-z0-9.\-]+\:.+$'
+    then
+        SSH_USER=$(echo "$DEST_FOLDER" | sed -E  's/^([A-Za-z0-9\._%\+\-]+)@([A-Za-z0-9.\-]+)\:(.+)$/\1/')
+        SSH_HOST=$(echo "$DEST_FOLDER" | sed -E  's/^([A-Za-z0-9\._%\+\-]+)@([A-Za-z0-9.\-]+)\:(.+)$/\2/')
+        SSH_DEST_FOLDER=$(echo "$DEST_FOLDER" | sed -E  's/^([A-Za-z0-9\._%\+\-]+)@([A-Za-z0-9.\-]+)\:(.+)$/\3/')
+        if [[ -n "$ID_RSA" ]] ; then
```
Comment (SimonHeimberg) [inline rsync_tmbackup.sh]: change unrelated to this commit (tab to spaces)

File: rsync_tmbackup.sh [hunk 8]
```diff
@@ -393,13 +393,13 @@ done
...
 fn_find_backup_marker() { fn_find "$(fn_backup_marker_path "$1")" 2>/dev/null; }
 
-if [ -z "$(fn_find_backup_marker "$DEST_FOLDER")" ]; then
-	fn_log_info "Safety check failed - the destination does not appear to be a backup folder or drive (marker file not found)."
-	fn_log_info "If it is indeed a backup folder, you may add the marker file by running the following command:"
-	fn_log_info ""
-	fn_log_info_cmd "mkdir -p -- \"$DEST_FOLDER\" ; touch \"$(fn_backup_marker_path "$DEST_FOLDER")\""
-	fn_log_info ""
-	exit 1
+if [[ -z "$(fn_find_backup_marker "$DEST_FOLDER")" ]]; then
```
Comment (SimonHeimberg) [inline rsync_tmbackup.sh]: change unrelated to this commit (tab to spaces)

**Comment by laurent22** [comment]
Yes thanks for pointing this out. I'm sorry but there are too many changes to be able to review this. And we use tabs, not spaces.
